### ISSUE TITLE
OUTPUT_DIR needs to be set for tests to run

### ISSUE THREAD

**Issue by chwilliams-ins** [FSMLang/FSMLang#125]
After cloning in WSL2, the `Linux.test` target fails. Setting `OUTPUT_DIR` fixes this but running `make Linux.test` from the `/src` directory should run without environment modification.

The `Linux` target works fine:
```
$ make Linux
make CFLAGS="-I ../src -I . " -C ../linux fsm
make[1]: Entering directory '/home/chwilliams/proj/FSMLang2/linux'
bison -d -y -v   ../src/parser.y
../src/parser.y: warning: 5 shift/reduce conflicts [-Wconflicts-sr]
mv -f y.tab.c parser.c
flex -I   -t ../src/lexer.l > lexer.c
rm lexer.c parser.c
bison -d -y -v   ../src/parser.y
../src/parser.y: warning: 5 shift/reduce conflicts [-Wconflicts-sr]
mv -f y.tab.c parser.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o parser.o parser.c
flex -I   -t ../src/lexer.l > lexer.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o lexer.o lexer.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o revision.o ../src/revision.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o fsm_utils.o ../src/fsm_utils.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o fsm_html.o ../src/fsm_html.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o fsm_plantuml.o ../src/fsm_plantuml.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o fsm_c.o ../src/fsm_c.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o fsm_c_common.o ../src/fsm_c_common.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o fsm_cswitch.o ../src/fsm_cswitch.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o list.o ../src/list.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o cwalk.o ../src/cwalk.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o ancestry.o ../src/ancestry.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o fsm_statistics.o ../src/fsm_statistics.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o fsm_c_event_xref.o ../src/fsm_c_event_xref.c
gcc -g -ggdb -Werror  -I ../src -I .  -DLINUX -DLINUX   -c -o fsm_rst.o ../src/fsm_rst.c
echo -I ../src -I .  -DLINUX -DLINUX
-I ../src -I . -DLINUX -DLINUX
gcc -g -ggdb -Werror  -o fsm  lexer.o parser.o revision.o fsm_utils.o fsm_html.o fsm_plantuml.o fsm_c.o fsm_c_common.o fsm_cswitch.o list.o cwalk.o ancestry.o fsm_statistics.o fsm_c_event_xref.o fsm_rst.o -lfl
make[1]: Leaving directory '/home/chwilliams/proj/FSMLang2/linux'
``` 

The environment doesn't have the OUTPUT_DIR var set:
```
$ echo $OUTPUT_DIR

```

The failure:
```
$ make Linux.test
make CFLAGS="-I ../src -I . " -C ../linux fsm
make[1]: Entering directory '/home/chwilliams/proj/FSMLang2/linux'
make[1]: 'fsm' is up to date.
make[1]: Leaving directory '/home/chwilliams/proj/FSMLang2/linux'
make -C ../linux ARCH=LINUX NEED_RINDEX= WEAK_FNS= test
make[1]: Entering directory '/home/chwilliams/proj/FSMLang2/linux'
make -C ../test clean
make[2]: Entering directory '/home/chwilliams/proj/FSMLang2/test'
make -C lexer clean
make[3]: Entering directory '/home/chwilliams/proj/FSMLang2/test/lexer'
../test.mk:15: /../depends.mk: No such file or directory
make[3]: *** No rule to make target '/../depends.mk'.  Stop.
make[3]: Leaving directory '/home/chwilliams/proj/FSMLang2/test/lexer'
make[2]: *** [Makefile:21: lexer.clean] Error 2
make[2]: Leaving directory '/home/chwilliams/proj/FSMLang2/test'
make[1]: *** [Makefile:32: test] Error 2
make[1]: Leaving directory '/home/chwilliams/proj/FSMLang2/linux'
make: *** [Makefile:40: Linux.test] Error 2
```

The fix:
```
$ pushd ../linux && export OUTPUT_DIR=${PWD}
~/proj/FSMLang2/linux ~/proj/FSMLang2/src
$ popd
```

Success:
```
$ make Linux.test
make CFLAGS="-I ../src -I . " -C ../linux fsm
make[1]: Entering directory '/home/chwilliams/proj/FSMLang2/linux'
make[1]: 'fsm' is up to date.
make[1]: Leaving directory '/home/chwilliams/proj/FSMLang2/linux'
make -C ../linux ARCH=LINUX NEED_RINDEX= WEAK_FNS= test
make[1]: Entering directory '/home/chwilliams/proj/FSMLang2/linux'
make -C ../test clean
make[2]: Entering directory '/home/chwilliams/proj/FSMLang2/test'
make -C lexer clean
make[3]: Entering directory '/home/chwilliams/proj/FSMLang2/test/lexer'
make[3]: Leaving directory '/home/chwilliams/proj/FSMLang2/test/lexer'
make -C parser clean
make[3]: Entering directory '/home/chwilliams/proj/FSMLang2/test/parser'
make[3]: Leaving directory '/home/chwilliams/proj/FSMLang2/test/parser'
make -C unit clean
make[3]: Entering directory '/home/chwilliams/proj/FSMLang2/test/unit'
make -C misc_cl_options CFLAGS="" clean
make[4]: Entering directory '/home/chwilliams/proj/FSMLang2/test/unit/misc_cl_options'
make[4]: Leaving directory '/home/chwilliams/proj/FSMLang2/test/unit/misc_cl_options'
```
Environment info:
```
$ grep DISTRIB_DESCRIPTION /etc/lsb-release && uname -srvm && git rev-parse
HEAD
DISTRIB_DESCRIPTION="Ubuntu 20.04.6 LTS"
Linux 5.15.167.4-microsoft-standard-WSL2 #1 SMP Tue Nov 5 00:21:55 UTC 2024 x86_64
05a770402174e9eaadbb17f2a7d7a2349e3cd207
```

**Comment by 9Fingered** [comment]
@chwilliams-ins - Thank you for your thorough issue report. The issue has been resolved; a PR is open.  The fix will be included in the upcoming 1.44 release.

**Comment by chwilliams-ins** [comment]
I mentioned WSL2 in the original issue report but I'm guessing this problem isn't limited to just that.
